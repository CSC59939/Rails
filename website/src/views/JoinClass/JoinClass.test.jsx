import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { JoinClass } from '..';

configure({ adapter: new Adapter() });

const sampleFunction = () => {

};

/* Snapshot */
describe('JoinClass Snapshot Test', () => {
  it('Render JoinClass without any error', () => {
    expect(renderer.create(<JoinClass />).toJSON()).toMatchSnapshot();
  });
  it('Render button without error', () => {
    const component = renderer.create(
      <JoinClass type="setting" onClick={sampleFunction} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


/* Enzyme */
const wrapper = shallow(<JoinClass />);
describe('JoinClass Enzyme Test', () => {
  it('Title tag should exists', () => {
    expect(wrapper.find('.title').closest('.join-class-page').length).toBe(1);
  });
  it('Join-card should exists', () => {
    expect(wrapper.exists('.join-card')).toBe(true);
  });
  it('Join-class-table should within the join-card', () => {
    expect(wrapper.find('.join-class-table').parent().is('.join-card')).toBe(true);
  });
});
