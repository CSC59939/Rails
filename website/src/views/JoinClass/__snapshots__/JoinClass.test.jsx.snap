// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinClass Render JoinClass without any eror 1`] = `
<div
  className="join-class-page"
=======
  className="join-class-page"
>>>>>>> do a small change
>
  <h1
    className="title"
  >
    Rails
  </h1>
  <div
    className="ant-card join-card ant-card-bordered"
=======
    className="ant-card join-card ant-card-bordered"
>>>>>>> do a small change
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Join Class
        </div>
=======
        <div
          className="ant-card-extra"
        >
          <a
            className="ant-btn ant-btn-danger"
            href="/dashboard"
            onClick={[Function]}
          >
            <span>
              Back to  Dashboard
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
<<<<<<< 2ebf9cacff8b7efc1af6bba5b68999698f8505d5
      <form
        className="ant-form ant-form-horizontal regiform"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        University/College
                      </div>
                      <div
                        className="ant-select-search ant-select-search--inline"
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      >
                        <div
                          className="ant-select-search__field__wrap"
                        >
                          <input
                            autoComplete="off"
                            className="ant-select-search__field"
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            value=""
                          />
                          <span
                            className="ant-select-search__field__mirror"
                          >

                            Â 
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <i
                        className="anticon anticon-down ant-select-arrow-icon"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-24"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection__placeholder"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        style={
                          Object {
                            "WebkitUserSelect": "none",
                            "display": "block",
                            "userSelect": "none",
                          }
                        }
                        unselectable="on"
                      >
                        Class and Section code
                      </div>
                    </div>
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <i
                        className="anticon anticon-down ant-select-arrow-icon"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          className="registerButton"
        >
          <button
            className="ant-btn ant-btn-primary"
            disabled={false}
            margin="auto"
            onClick={[Function]}
            type="button"
          >
            <span>
              Join Class
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
  <br />
      <div
        className="ant-select-lg ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          aria-autocomplete="list"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "block",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              University/College
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >

                  Â 
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
      <div
        className="ant-table-wrapper join-class-table"
      >
        <div
          className="ant-spin-nested-loading"
          style={null}
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
              style={Object {}}
            >
              <div
                className="ant-table-content"
              >
                <div
                  className="ant-table-body"
                  onScroll={[Function]}
                  onWheel={[Function]}
                  style={Object {}}
                >
                  <table
                    className=""
                    style={Object {}}
                  >
                    <colgroup>
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr
                        style={
                          Object {
                            "height": null,
                          }
                        }
                      >
                        <th
                          className=""
                        >
                          <div
                            onClick={[Function]}
                            title="Sort"
                          >
                            Class Code
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div
                            onClick={[Function]}
                            title="Sort"
                          >
                            Class Name
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div
                            onClick={[Function]}
                            title="Sort"
                          >
                            Instructor
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div
                            onClick={[Function]}
                            title="Sort"
                          >
                            Time
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div
                            onClick={[Function]}
                            title="Sort"
                          >
                            Days
                          </div>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  className="ant-table-placeholder"
                >
                  No data
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
